app/models/saved_claim.rb

```diff
@@ -6,17 +6,32 @@ class SavedClaim < ActiveRecord::Base
   validate(:form_must_be_string)
   attr_encrypted(:form, key: Settings.db_encryption_key)
 
-  before_create do
-    self.guid = SecureRandom.uuid
-    self.form_type = self.class::FORM.upcase
+  has_many :persistent_attachments
+
+  after_initialize do
+    self.guid ||= SecureRandom.uuid
+    self.form_id = self.class::FORM.upcase
+  end
+
+  def process_attachments!
+    GenerateClaimPDFJob.perform_async(id) if respond_to?(:to_pdf)
+    refs = attachment_keys.map { |key| Array(open_struct_form.send(key)) }.flatten
+    files = PersistentAttachment.where(guid: refs.map(&:confirmationCode))
+    files.update_all(saved_claim_id: id)
+    files.reload.each.map(&:process)
```

Comment:
`process` just kicks off a Sidekiq worker per file, so there's no real opportunity to fix everything with a rollback.